src/core/server/http/http_server.ts

```diff
@@ -87,7 +90,11 @@ export class HttpServer {
   private readonly authRequestHeaders: AuthHeadersStorage;
   private readonly authResponseHeaders: AuthHeadersStorage;
 
-  constructor(private readonly logger: LoggerFactory, private readonly name: string) {
+  constructor(
+    private readonly logger: LoggerFactory,
+    private readonly name: string,
+    private readonly gracefulShutdownTimeout$: Observable<Duration>
```

Comment:
Can't we just `await` this value in the service and then pass it down instead of propagating an observable everywhere?

Reply:
or can you use the sync API?
```typescript
configService.atPathSync<HttpConfigType>(httpConfig.path).gracefulShutdownTimeout
```

Reply:
I thought of doing that, however, since the value only applies to the final shutdown of Kibana, I went for the Observable to save us from some work whenever we implement hot reloading of the config. Another reason I went for the Observable is that the HTTP service also relies on the async API, so I thought it'd provide a more consistent UX.

IMO, it doesn't really affect anything: `stop` was already a promise (and it will always be to allow graceful shutdowns), and the NotReady server is running with the first version of the config.

If you think the sync API should be used instead, I'm happy to change it. Although I wonder if we should also change it in the rest of the HTTP service then.

Reply:
> I went for the Observable to save us from some work whenever we implement hot reloading of the config.

It's already implemented, isn't it? But we don't aim to support it for *all* the config values.

> Although I wonder if we should also change it in the rest of the HTTP service then.

Are there any places when `config$` leaks outside of `http_service`? Config in `http_server` is passed as POJO https://github.com/elastic/kibana/blob/265cc7624768fef99a7cad9045a79e8610308886/src/core/server/http/http_server.ts#L109

Reply:
Makes sense, I've moved to the sync API.